openwisp_monitoring/device/schema.py

```diff
@@ -35,7 +35,20 @@
                         "total": {"type": "integer"},
                         "free": {"type": "integer"},
                         "buffered": {"type": "integer"},
-                        "cache": {"type": "integer"},
+                        "cached": {"type": "integer"},
+                        "shared": {"type": "integer"},
+                        "available": {"type": "integer"},
+                    },
+                },
+                "disk": {
+                    "type": "array",
```

Comment:
this schema is wrong. Please see the definition of other arrays and check the jsonschema spec.

Please flag the fields required to do the calculation of the value we store as required.

Please also write a test which sends bogus disk data to ensure the API rejects it.

Reply:
> Please flag the fields required to do the calculation of the value we store as required.

I didn't get this.

Reply:
> > Please flag the fields required to do the calculation of the value we store as required.
> 
> I didn't get this.

To flag as required all the fields that are needed for the calculation of the used disk percentage which we graph, otherwise we would be allowing to send data which will not work or may even cause exceptions.

I see that you flagged them all required so that's ok for me.

Reply:
Define the fields of the disk section that you use to do the calculations as required, you did it.

Reply:
That's already done. I have made all fields required taking into account that all shall be needed for #116